src/lower/compressed_mode_type.cpp

```diff
@@ -195,7 +209,7 @@ Expr CompressedModeType::getPosCapacity(Mode& mode) const {
   return mode.getVar(varName);
 }
 
-Expr CompressedModeType::getIdxCapacity(Mode& mode) const {
+Expr CompressedModeType::getIdxCapacity(Mode mode) const {
   const std::string varName = mode.getName() + "_idx_size";
```

Comment:
Perhaps this should be renamed `_coord_size` if the `idx` array is to be renamed to `coord`?